python/ray/serve/schema.py

```diff
@@ -413,13 +413,21 @@ class DeploymentSchema(BaseModel, allow_population_by_field_name=True):
 
     @root_validator
     def num_replicas_and_autoscaling_config_mutually_exclusive(cls, values):
-        if values.get("num_replicas", None) not in [DEFAULT.VALUE, None] and values.get(
-            "autoscaling_config", None
-        ) not in [DEFAULT.VALUE, None]:
-            raise ValueError(
-                "Manually setting num_replicas is not allowed "
-                "when autoscaling_config is provided."
-            )
+        num_replicas = values.get("num_replicas", None)
+        autoscaling_config = values.get("autoscaling_config", None)
+
+        # Cannot have `num_replicas` be an int and a non-null
+        # autoscaling config
+        if isinstance(num_replicas, int):
+            if autoscaling_config not in [None, DEFAULT.VALUE]:
+                raise ValueError(
+                    "Manually setting num_replicas is not allowed "
+                    "when autoscaling_config is provided."
+                )
+        # A null `num_replicas` or `num_replicas="auto"` can be paired
+        # with a non-null autoscaling_config
+        elif num_replicas not in ["auto", None, DEFAULT.VALUE]:
+            raise ValueError(f"Invalid value for `num_replicas`: {num_replicas}")
```

Comment:
This error isn't actionable, be sure to specify what it can be set to
```suggestion
            raise ValueError(f"`num_replicas` must be an int or 'auto', but got: {num_replicas}")
```